lib/api.js

```diff
@@ -282,117 +264,77 @@ api.declare({
   return res.reply(result);
 });
 
-function createPubKeyHash(pubKey) {
-  assert(typeof pubKey === 'string');
-  let keyData = pubKey.split(' ');
-  assert(keyData.length >= 2, 'pub key must be in a valid format');
-  keyData = keyData[0] + ' ' + keyData[1];
-  keyData = crypto.createHash('sha256').update(keyData).digest('hex');
-  return keyData.slice(0, 7);
-};
-
-function createKeyPairName(prefix, pubKey, workerName) {
-  assert(typeof prefix === 'string');
-  // We want to support the case where we're still using a config setting
-  // that ends in : as it used to
-  if (prefix.charAt(prefix.length - 1) === ':') {
-    prefix = prefix.slice(0, prefix.length - 1);
-  }
-  assert(prefix.indexOf(':') === -1, 'only up to one trailing colon allowed');
-  assert(typeof pubKey === 'string');
-  assert(typeof workerName === 'string');
-  return prefix + ':' + workerName + ':' + createPubKeyHash(pubKey);
-};
-
-function parseKeyPairName(name) {
-  assert(typeof name === 'string');
-  let parts = name.split(':');
-  assert(parts.length === 3, 'Unparsable keypair name: ' + name);
-  return {
-    prefix: parts[0],
-    workerType: parts[1],
-    keyHash: parts[2],
-  };
-};
-
 api.declare({
   method: 'get',
-  route: '/worker-types/:workerType/key-pair',
+  route: '/key-pairs/:name',
   name: 'ensureKeyPair', 
   title: 'Ensure a KeyPair for a given worker type exists',
   stability: API.stability.experimental,
   deferAuth: true,
-  scopes: [['ec2-manager:manage-resources:<workerType>']],
+  input: 'create-key-pair.json#',
+  scopes: [['ec2-manager:manage-key-pairs:<name>']],
   description: [
     'Ensure that a keypair of a given name exists.  This call caches',
     'internally the list of keypair names it has ensured at least one',
     'time, and as such is safe to call repeatedly.  It is idempotent.',
```

Comment:
the code that caches known keypairs appears to have been removed, so this description should be adjusted accordingly.